library/common/extensions/retry/options/network_configuration/predicate.cc

```diff
@@ -0,0 +1,25 @@
+#include "library/common/extensions/retry/options/network_configuration/predicate.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Options {
+
+NetworkConfigurationRetryOptionsPredicate::NetworkConfigurationRetryOptionsPredicate(
+    const envoymobile::extensions::retry::options::network_configuration::
+        NetworkConfigurationOptionsPredicate&,
+    Upstream::RetryExtensionFactoryContext& context) {
+  network_configurator_ = Network::NullableConfiguratorHandle{context.singletonManager()}.get();
+  RELEASE_ASSERT(network_configurator_ != nullptr, "unexpected nullptr network configurator");
```

Comment:
I made this an assert given that it would be a programming error for the singleton to not be present at the time we build the retry predicate.

Reply:
Yep, this was my thinking as well.